copado-function/app/Deploy.js

```diff
@@ -255,10 +255,45 @@ async function run() {
         throw ex;
     }
 
+    let gitAddArr;
+    try {
+        Log.info('');
+        Log.info('Retrieve components');
+        Log.info('===================');
+        Log.info('');
+        gitAddArr = await Commit.retrieveCommitSelection(targetBU, commitSelectionArr);
+    } catch (ex) {
+        Log.error('Retrieving failed: ' + ex.message);
+        Copado.uploadToolLogs();
+        throw ex;
+    }
+
+    try {
+        Log.info('');
+        Log.info('Add components in metadata JSON to Git history');
+        Log.info('===================');
+        Log.info('');
+        Commit.addSelectedComponents(gitAddArr);
+    } catch (ex) {
+        Log.error('git add failed:' + ex.message);
+        throw ex;
+    }
+
+    try {
+        Log.info('');
+        Log.info('Commit components');
+        Log.info('===================');
+        Log.info('');
+        Commit.commit(CONFIG.mainBranch);
+    } catch (ex) {
+        Log.error('git commit failed:' + ex.message);
+        throw ex;
+    }
+
```

Comment:
you might want to move this section for the new commit into a new method in Deploy() to make this easier to read & understand